src/main/java/sirius/biz/storage/s3/ObjectStore.java

```diff
@@ -726,6 +731,8 @@ public void upload(BucketName bucket, String objectId, InputStream inputStream)
                                       eTags.size());
             }
 
+            emptyInputStream = eTags.isEmpty();
+
             getClient().completeMultipartUpload(new CompleteMultipartUploadRequest(bucket.getName(),
```

Comment:
could we just abort the empty upload here? has we know for sure that an empty MP upload is pointless anyway? could be use a conventional upload to ensure that the empty object exists?

Reply:
maybe we could just call

```
getClient().abortMultipartUpload(new AbortMultipartUploadRequest(bucket.getName(),
                                                                             objectId,
                                                                             multipartUpload.getUploadId()));
                                                                             ```
                                                                             
                                                                             followed by s.th. like
```
getClient().putObject(bucket.getName(),
                                  objectId,"");
```

Reply:
In the replication event is indeed pointless. But a "normal" upload is the real question: shall we always assume no data as OK? If so, we should than load a 0-byte file. Aborting the transaction means the object id does not exist in the bucket.
Inside ObjectStore, we cannot know that the space is being used for replication (one can even have the idea to chain replications)... 
This is why I decided to keep it non-breaking, preserving the previous behavior and only catch the case when executing a replication.